### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yelnikov-andrii**

### CODE REVIEW

**Review by v-vytrykush — COMMENTED** [review]
Great job! 
Also you can give variables 'a' and 'i' more clearer names.

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,22 @@
...
+      for (let a = 0; a < actions[i].keysToRemove.length; a++) {
+        delete state[actions[i].keysToRemove[a]];
+      }
+    }
 
+    if (actions[i].type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
```
Comment (Denchepornyuk) [inline src/transformState.js:23]: use `switch case` instead of `if`

**Review by denys-cheporniuk — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,21 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties': Object.assign(state, actions[i].extraData);
```
Comment (Denchepornyuk) [inline src/transformState.js:10]:
```suggestion
      case 'addProperties':
        Object.assign(state, actions[i].extraData);
```

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,21 @@
...
+        break;
+
+      case 'removeProperties':
+        for (let a = 0; a < actions[i].keysToRemove.length; a++) {
+          delete state[actions[i].keysToRemove[a]];
+        }
+        break;
 
+      case 'clear': for (const key in state) {
```
Comment (Denchepornyuk) [inline src/transformState.js:19]:
```suggestion
      case 'clear':
        for (const key in state) {
```

**Review by denys-cheporniuk — APPROVED** [review]

**Pull request closed without merging**